### PR TITLE
Provides a default configmap

### PR DESCRIPTION
**Pull request by karmab**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
provides a default configmap with LiveMigration and DataVolumes enabled
```

### CODE REVIEW

File: manifests/release/kubevirt.yaml.in [hunk 1]
```diff
@@ -5,6 +5,14 @@ metadata:
...
     kubevirt.io: ""
   name: {{.Namespace}}
+---
+apiVersion: v1
+data:
+  feature-gates: LiveMigration,DataVolumes
+kind: ConfigMap
+metadata:
+  name: kubevirt-config
+  namespace: kubevirt
```
Comment (slintes) [inline manifests/release/kubevirt.yaml.in]: please use {{.Namespace}}

**Comment by davidvossel** [comment]
> 'm not sure if we want this, it will probably break workflow where users create the CM themself

right, we don't want this config map in our release manifests. 

We could consider having the operator create this configmap during the install process, and expose the config-maps variables in the KubeVirt cr... so users would define what feature gates they want opened when they post the KubeVirt crd to the cluster.

**Comment by karmab** [comment]
ok, i will close the PR then but note that 
 - most users will want datavolumes and live migrations 
- they will know they want it at install time
 - they will have to deploy, create the configmap and then delete the pods from the kubevirt namespace. Not an ideal workflow imho...

**Comment by davidvossel** [comment]
> Users will get datavolumes and live migrations by default - once they are
ready.

I'd like to depreciate the datavolume feature flag entirely at this point. The issue is dynamic detection of the datavolume CRD. We need to create datavolume informers at initialization time for our controllers. If we attempt to detect the CRD at initialization, then we have an ordering issue where kubevirt might launch without datavolume support because of the timing involved with when CDI was installed.

I opened #1638  awhile back. We need to either reinitialize our controller once the DataVolume crd is introduced, or create a dynamic informer that can magically start working after the CRD is created without requiring reinitialization.

**Comment by fabiand** [comment]
+1 And this is something we probably want to adopt for other feature flags
as well. IOW an operator could expect that the system autoamtically
supports changed configuration.

- fabian

**Pull request closed without merging**